### PR TITLE
Validate context length for generation and validation

### PR DESCRIPTION
**Pull request by sfc-gh-jhilgart**

### CODE REVIEW

File: semantic_model_generator/validate/context_length.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+_MODEL_CONTEXT_LENGTH = 7000  # We use 7k so that we can reserve 1k for response tokens.
```
Comment (sfc-gh-nsehrawat) [inline semantic_model_generator/validate/context_length.py:1]: Shouldn't we account for instruction tokens in the prompt?

Comment (sfc-gh-jhilgart) [reply semantic_model_generator/validate/context_length.py:1]: This is pretty loose as is. Instruction tokens, even for llama3, is only ~20 tokens. 

I can add an additional buffer here though!

File: semantic_model_generator/validate/context_length.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
```
Comment (sfc-gh-dasilva) [inline semantic_model_generator/validate/context_length.py:1]: We should reduce this to also account for the tokens consumed by our prompts

**Review by sfc-gh-dasilva — APPROVED** [review]

File: semantic_model_generator/validate/context_length.py (hunk 2 above)
Comment (sfc-gh-jhilgart) [reply semantic_model_generator/validate/context_length.py]: great call - updated

**Pull request merged**